device/device.go

```diff
@@ -245,32 +245,10 @@ func (d *Device) IsConnected() bool {
 }
 
 // AddInterface adds an interface to the device. The interface must be loaded with ParseInterface
-// from the astarte-go/interfaces package.
+// from the astarte-go/interfaces package: when parsed, the interface is also validated.
 // AddInterface returns `nil` if the interface was loaded successfully, or a corresponding error
 // otherwise (e.g. interface validation failed).
 func (d *Device) AddInterface(astarteInterface interfaces.AstarteInterface) error {
-	if err := astarteInterface.Aggregation.IsValid(); err != nil {
-		return err
-	}
-	if err := astarteInterface.Type.IsValid(); err != nil {
-		return err
-	}
-	if err := astarteInterface.Ownership.IsValid(); err != nil {
-		return err
-	}
-
-	for _, mapping := range astarteInterface.Mappings {
-		if err := mapping.Reliability.IsValid(); err != nil {
-			return err
-		}
-		if err := mapping.Retention.IsValid(); err != nil {
-			return err
-		}
-		if err := mapping.DatabaseRetentionPolicy.IsValid(); err != nil {
-			return err
-		}
-	}
-
 	d.interfaces[astarteInterface.Name] = astarteInterface
 	return nil
```

Comment:
Now this method never returns an error, so it would make sense to change its return type. On the other side, this would break the API, but since we are still in a `0.x` release this might be allowed - if properly documented. What do we do?

Reply:
I agree that it is useless returning an error (which is always `nil`). It is meant not to change the current APIs. We should agree on how to proceed (cc @rbino)